src/llamacrawl/services/readers/firecrawl.py

```
@@ -1125,22 +1141,25 @@ async def _async_crawl_with_watcher(
                 progress_callback=progress_callback,
             )
 
+        watcher_doc_types = [type(d).__name__ for d in collected_docs[:3]] if collected_docs else []
         self.logger.debug(
             f"Watcher crawl completed with {len(collected_docs)} total documents",
```

Ruff (0.14.0)

Logging statement uses f-string

(G004)